person.rb

```diff
@@ -0,0 +1,54 @@
+class Person
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    @id = 1
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```

Comment:
- [ ] By using this way will probably throw an error saying: NoMethodError: undefined method `id, name, age and parent_permission`.

`NoMethodError` is the error you get when you ask for the value of `age, name and parent_permission` without the proper setup.

The solution, you can define your own method like this:
class Person 
  def name
    @name
  end
end

However, this should be hard/complicated while we need to implement more methods.

So, there is a better way to define this kind of method? like `attr_accessor`, `attr_reader` and `attr_writer`.

class Person
 attr_accessor :name
 def initialize(name)
   @name = name
 end
end

Reply:
Done